crates/sui-types/src/object.rs

```diff
@@ -216,10 +216,26 @@ impl MoveObject {
         self.contents.splice(ID_END_INDEX.., value.to_le_bytes());
     }
 
+    /// Update the `timestamp_ms: u64` field of the `Clock` type.
+    ///
+    /// Panics if the object isn't a `Clock`.
+    pub fn set_clock_timestamp_ms_unsafe(&mut self, timestamp_ms: u64) {
```

Comment:
Instead of adding this function, you could also add a test object constructor that constructs a clock object with arbitrary time value? (more aligned with how we create test coins)

Reply:
The use of this function is to initialize the clock in genesis, i suppose i could look at improving the constructor in move for that, but i initially thought that we'd hit a compat issue there